src/App.tsx

```diff
@@ -3,12 +1,29 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { TodoList } from './components/TodoList';
+import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todosFromServer);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodosFromServer)
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const selectedTodo = filteredTodos.find(todo => todo.id === selectedTodoId);
```

Comment:
This operation of finding a todo is being performed on each render which can be inefficient if the list is large. Consider memoizing `selectedTodo` with `useMemo`.